spacemeshos/smapp
# fix: Stop smeshing clean up smeshing-proving-opts in config

It closes #1385 

### How to test:
<img width="292" alt="Screenshot 2023-07-21 at 17 36 57" src="https://github.com/spacemeshos/smapp/assets/54288612/59201f41-9abd-4da5-8d55-46f1c0ada822">

**A:** missing smeshing-proving-opts 
1. open custom config and remove smeshing-proving-opts ( init the smeshing before )
3. sign in 
4. see the modal
5.  update settings 
6. settings should be updated

**B:** update smeshing-proving-opts 
1. start smashing 
2. click edit 
3. edit PoS proving opts
4. see the smeshing update  
5. settings should be updated

**C:** update smeshing-proving-opts  with enabled auto-start
1. enable auto-start
2. restart PC
3. see the opened app and the modal with update proving settings
4. update settings
5. settings should be updated

**D:** update smeshing-proving-opts  with the broken node by default ( for example remove the boot nodes from the config )
1. start the app
2. node is down
3. see  modal with update proving settings
4. update settings
5. settings should be updated 

<img width="923" alt="Screenshot 2023-07-21 at 16 59 02" src="https://github.com/spacemeshos/smapp/assets/54288612/54bb38e9-0f28-4ea7-bcf4-dc6b2ca45332">
<img width="986" alt="Screenshot 2023-07-21 at 16 59 13" src="https://github.com/spacemeshos/smapp/assets/54288612/c0b20fcf-3362-4ef1-9ab5-129e6c425672">

<img width="900" alt="Screenshot 2023-07-21 at 17 05 25" src="https://github.com/spacemeshos/smapp/assets/54288612/5bb74dda-17ff-4613-a6c1-f87eba415418">
<img width="152" alt="Screenshot 2023-07-21 at 16 59 25" src="https://github.com/spacemeshos/smapp/assets/54288612/1273e65a-176c-441a-a930-e03ca6a01d2e">



<img width="1100" alt="Screenshot 2023-07-21 at 17 43 48" src="https://github.com/spacemeshos/smapp/assets/54288612/ab2c3d16-161f-40a1-ad80-1c4612d4f6d8">

###  Checklist
need to test and resolve additional cases from the code 
- [ ] wallet only
- [ ] auto-start
